### PR TITLE
Clean up AnnotatedTypeMirror: remove commented out code and actually

### PR DESCRIPTION
**Pull request by wmdietl**
prevent adding annotations to AnnotatedExecutableType.

### CODE REVIEW

File: checker/src/org/checkerframework/checker/guieffect/GuiEffectTypeFactory.java [hunk 1]
```diff
@@ -469,16 +469,16 @@ public boolean hasExplicitEffect(ExecutableElement methElt) {
         public Void visitMethod(MethodTree node, AnnotatedTypeMirror type) {
             AnnotatedTypeMirror.AnnotatedExecutableType methType =
                     (AnnotatedTypeMirror.AnnotatedExecutableType) type;
-            Effect e = getDeclaredEffect(methType.getElement());
+            // Effect e = getDeclaredEffect(methType.getElement());
```
Comment (smillst) [inline checker/src/org/checkerframework/checker/guieffect/GuiEffectTypeFactory.java]: This doesn't seem to be related to cleaning up AnnotatedTypeMirror.  Nor does the other change to the GUI Effect Checker.

Comment (wmdietl) [reply checker/src/org/checkerframework/checker/guieffect/GuiEffectTypeFactory.java]: They are: on line 480 in this file and on line 243 in the other file an annotation is added to an `AnnotatedExecutableType`.
Neither of these assignments had any effects, so I commented out this practically dead code.
The change in ATM will start throwing an exception for these calls, which is the reason why I commented them out.
No other code used `e`, so I also commented that out.
There already was some commented-out code in both places. Would you prefer completely removing this code?

Comment (smillst) [reply checker/src/org/checkerframework/checker/guieffect/GuiEffectTypeFactory.java]: Oh, I see now.  The pull request comment didn't mention anything about commenting out dead code, so I was confused.
